### PR TITLE
[IDED] Syndicate shuttle .50 BMG turrets' bullets now self-disintegrate before hitting a nukie

### PR DESCRIPTION
**Pull request by carlarctg**
## About The Pull Request

image
## Why It's Good For The Game

The amount of times these have caused my death as a nukie op is crazy. I see it all the time, either a nukie takes damage or a cyborg does. Last Terry round TWO CYBORGS (one was me) lost all their modules and were left to flounder around helplessly because this dumb fuck turret noticed some jackass through a wall and annihilated our health.

These things are a huge hassle and 95% of the time some funnyman comes in with smart foam to hard counter them anyways. At least let them not be a net negative to nuclear operatives.
## Changelog
:cl:
balance: Syndicate shuttle .50 BMG turrets' bullets now self-disintegrate before hitting a nukie
/:cl:

### CODE REVIEW

**Comment by Pickle-Coding** [comment]
Don't like this and it's anti-sandbox. Getting shot is a consequence of the bullet colliding with you. If you don't want to get shot, avoid the bullet colliding with you. Get out of the way. I haven't observed this phenomenom being a major issue for operatives either. Never heard of anyone report this until now, so I don't believe it's a significant concern unless there's evidence to the contrary.

**Comment by Loafin34** [comment]
turn off the turrets

File: code/modules/projectiles/projectile/bullets/sniper.dm [hunk 1]
```diff
@@ -90,6 +90,17 @@
 	speed = 0.3
 	range = 16
 
+/obj/projectile/bullet/p50/penetrator/shuttle/prehit_pierce(atom/target)
+	. = ..()
+	if(!isliving(target))
+		return .
+
+	var/mob/living/our_target = target
+	if(IS_NUKE_OP(our_target))
```
Comment (Fikou) [inline code/modules/projectiles/projectile/bullets/sniper.dm:99]: no direct antag checks find a better method

**Review by Fikou — CHANGES_REQUESTED** [review]

**Comment by SomeoneYouProbablyKnow** [comment]
Yea... too bad? This would allow for what's essentially easier turret camping. Turrets are already plenty unfun if you dont have the right gear to take em on.
If they are a net negative, turn em off. The one time this might be a reasonable change might the the assault drop pod.

File: code/modules/projectiles/projectile/bullets/sniper.dm (hunk 1 above)
Comment (carlarctg) [reply code/modules/projectiles/projectile/bullets/sniper.dm:99]: ur mom

**Comment by mc-oofert** [comment]
make them just avoid firing if theres a nuclear operative in the way problem solved

**Comment by moocowswag** [comment]
They should have a manual operation mode and limited ammo (or require power to self fabricate ammo)

**Comment by carlarctg** [comment]
> make them just avoid firing if theres a nuclear operative in the way problem solved

is there a meaningful difference here tho

**Comment by Jacquerel** [comment]
> > make them just avoid firing if theres a nuclear operative in the way problem solved
> 
> is there a meaningful difference here tho

An extremely significant one, which is that dissolving bullets is stupid and a turret not firing when it detects friendlies in the way makes sense.

**Comment by Kapu1178** [comment]
ided

**Comment by Redrover1760** [comment]
> They should have a manual operation mode and limited ammo (or require power to self fabricate ammo)
> 
> i would also argue that certain nukie armors should have way more ballistic protection than even laser or melee prot since thats what they specialize in while nanotrasen specializes in laser and anti laser tech.

They already do. Elite nukie suit gets bullet 6 and nukie suit gets bullet 5, which is their best protection stats. Good for mitigating friendly fire.

**Comment by carlarctg** [comment]
will use los check whwen my other pr is meged

**Comment by carlarctg** [comment]
if both maints and players dont liek it then theres no point

**Pull request closed without merging**